src/app/tabs/newAdmin/resolver/operations.js

```diff
@@ -101,6 +101,93 @@ export const getContentBytes = (resolverAddress, domain) => (dispatch) => {
     .catch(error => dispatch(errorContent(CONTENT_BYTES, error)));
 };
 
+const updateTextRecordToLocalStorage = (domain, key, add = false) => {
+  const storedKeys = localStorage.getItem('keys')
+    ? JSON.parse(localStorage.getItem('keys')) : {};
+  if (!storedKeys[domain]) {
+    storedKeys[domain] = [];
+  }
+  const eipKeys = ['email', 'url', 'avatar', 'description', 'notice', 'keywords', 'com.discord', 'com.github', 'com.reddit', 'com.twitter ', 'org.telegram'];
+  if (!eipKeys.includes(key)) {
+    if (add) {
+      if (storedKeys[domain].indexOf(key) === -1) {
+        storedKeys[domain].push(key);
+      }
+    } else {
+      storedKeys[domain].pop(key);
+    }
+    localStorage.setItem('keys', JSON.stringify(storedKeys));
+  }
+};
+const removeTextRecordFromLocalStorage = (domain, key, remove = false) => {
+  const storedKeys = localStorage.getItem('keys')
+    ? JSON.parse(localStorage.getItem('keys')) : {};
+  if (!storedKeys[domain]) {
+    storedKeys[domain] = [];
+  }
+  const index = storedKeys[domain].indexOf(key);
+  const eipKeys = ['email', 'url', 'avatar', 'description', 'notice', 'keywords', 'com.discord', 'com.github', 'com.reddit', 'com.twitter ', 'org.telegram'];
+  if (!eipKeys.includes(key)) {
+    if (remove) {
+      if (index > -1) {
+        storedKeys[domain].splice(index, 1);
+      }
+    }
+  }
+  localStorage.setItem('keys', JSON.stringify(storedKeys));
+};
+/**
+ * Querys the blockchain for the assosiated text records keys and returns values
+ * @param {address} resolverAddress address of the domain's resolver
+ * @param {domain} domain domain associated with the text record.
+ */
+export const getTextRecord = (resolverAddress, domain, value) => async (dispatch) => {
+  dispatch(requestContent(TEXT_RECORD));
+  const storedKeys = localStorage.getItem('keys')
+    ? JSON.parse(localStorage.getItem('keys')) : {};
+  const hash = namehash(domain);
+  const web3 = new Web3(window.rLogin);
```

Comment:
This is also duplicated on this page, let's create a single instance for it.

Reply:
Change addressed:
```
export const getLocalStoredKeys = () => {
  const storedKeys = localStorage.getItem('keys')
    ? JSON.parse(localStorage.getItem('keys')) : {};
  return storedKeys;
};
```